contracts/secretswap_router/src/contract.rs

```diff
@@ -0,0 +1,378 @@
+use cosmwasm_std::{
+    debug_print, from_binary, to_binary, Api, Binary, Coin, CosmosMsg, Env, Extern, HandleResponse,
+    InitResponse, Querier, StdError, StdResult, Storage, WasmMsg,
+};
+use secret_toolkit::snip20;
+use secretswap::{Asset, AssetInfo};
+use HandleMsg::RecoverFunds;
+
+use crate::{
+    msg::{HandleMsg, Hop, InitMsg, NativeSwap, QueryMsg, Route, Snip20Swap},
+    state::{
+        delete_route_state, read_owner, read_route_state, read_tokens, store_owner,
+        store_route_state, store_tokens, RouteState,
+    },
+};
+
+pub fn init<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    _msg: InitMsg,
```

Comment:
Would probably be easier to accept a list of tokens here and register them all at once instead of calling a separate function